src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+        if ((player->currentMask == PLAYER_MASK_BREMEN) && (globalCtx->interfaceCtx.unk_21E == 0x1A)) {
+            return ITEM_F1;
+        }
+
+        if ((player->currentMask == PLAYER_MASK_KAMARO) && (globalCtx->interfaceCtx.unk_21E == 0x19)) {
+            return ITEM_F2;
+        }
+
+        return item;
+    }
+
+    if (arg2 == 1) {
+        return (gSaveContext.buttonStatus[1] != BTN_DISABLED) ? gSaveContext.equips.buttonItems[0][1]
+               : (gSaveContext.unk_3F22 == 0x10)              ? gSaveContext.equips.buttonItems[0][1]
+                                                              : ITEM_NONE;
+    }
+
+    if (arg2 == 2) {
+        return (gSaveContext.buttonStatus[2] != BTN_DISABLED) ? gSaveContext.equips.buttonItems[0][2]
+               : (gSaveContext.unk_3F22 == 0x10)              ? gSaveContext.equips.buttonItems[0][2]
+                                                              : ITEM_NONE;
+    }
+
+    return (gSaveContext.buttonStatus[3] != BTN_DISABLED) ? gSaveContext.equips.buttonItems[0][3]
+           : (gSaveContext.unk_3F22 == 0x10)              ? gSaveContext.equips.buttonItems[0][3]
+                                                          : ITEM_NONE;
+}
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123810.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123960.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801239AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123AA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123BD4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123C58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_SetEquipmentData.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123D50.s")
+extern u8 sActionModelGroups[];
+
+s32 Player_ActionToModelGroup(Player* player, s32 actionParam) {
+    s32 modelGroup = sActionModelGroups[actionParam];
+
+    if ((modelGroup == 2) && func_801234B0(player)) {
+        return 1;
+    }
+    return modelGroup;
+}
+
+extern u8 D_801BFF90[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF90[PLAYER_FORM_MAX] = {
+    PLAYER_BOOTS_FIERCE_DEITY, // PLAYER_FORM_FIERCE_DEITY
+    PLAYER_BOOTS_GORON,        // PLAYER_FORM_GORON
+    PLAYER_BOOTS_ZORA_LAND,    // PLAYER_FORM_ZORA
+    PLAYER_BOOTS_DEKU,         // PLAYER_FORM_DEKU
+    PLAYER_BOOTS_HYLIAN,       // PLAYER_FORM_HUMAN
+};
+#endif
+
+extern u8 D_801BFF98[PLAYER_FORM_MAX];
+#if 0
+u8 D_801BFF98[PLAYER_FORM_MAX] = {
+    1, // PLAYER_FORM_FIERCE_DEITY
+    3, // PLAYER_FORM_GORON
+    2, // PLAYER_FORM_ZORA
+    0, // PLAYER_FORM_DEKU
+    1, // PLAYER_FORM_HUMAN
+};
+#endif
+
+typedef struct {
+    /* 0x00 */ u8 flag;
+    /* 0x02 */ u16 textId;
+} TextTriggerEntry; // size = 0x04
+
+extern TextTriggerEntry D_801BFFA0[];
+#if 0
+TextTriggerEntry D_801BFFA0[] = {
+    { 1, 0x26FC },
+    { 2, 0x26FD },
+    { 0, 0 },
+    { 2, 0x26FD },
+};
```

Comment:
What are these textIds? We'd be a long way to understanding this if we had that information.

Reply:
Those textIds are not present on the text dump I have

Reply:
Hmm, there don't seem to be any `0x26..` ones... are we sure that's what they are?

Reply:
Yes, I am. This is only used on `func_801242DC` (OoT's: `func_8008F2F8`), and that value is passed to `func_801518B0`, which is `Message_StartTextbox`.

Reply:
Oh, I see. It's for the environment text. So probably they were going to keep them at some point, but they got dropped. Can explain that... not sure what would happen if it was used, but this probably is not even touched (or presumably you'd get UB).